### PR TITLE
Support dlsource as an attribution parameter (fixes #159)

### LINKED ISSUE #159: support `dlsource` as part of attribution data
As part of https://bugzilla.mozilla.org/show_bug.cgi?id=1814727 we'll be introducing a new *dlsource* key to attribution information. We'll be asking Bedrock to set this key to *mozorg* for any downloads initiated there, and the attribution service will need to be able to ensure it ends up in the build.

### PR DESCRIPTION
**Pull request by bhearsum**
@willdurand - I think this covers most, if not everything? I tried to use the stubtestclient locally, but wasn't able to make it work (I presume I need some GCS or AWS creds set-up...)

### CODE REVIEW

**Comment by willdurand** [comment]
> I tried to use the stubtestclient locally, but wasn't able to make it work (I presume I need some GCS or AWS creds set-up...)

That you cannot successfully call -dev or -stage makes sense but you could use the local server, right?

File: attributioncode/validator.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var validAttributionKeys = map[string]bool{
 	"visit_id":       true, // https://bugzilla.mozilla.org/show_bug.cgi?id=1677497
 	"session_id":     true, // https://bugzilla.mozilla.org/show_bug.cgi?id=1809120
 	"client_id":      true, // Alias of `visit_id`.
+	"dlsource":       true, // https://github.com/mozilla-services/stubattribution/issues/159
```
Comment (willdurand) [inline attributioncode/validator.go:41]: Nit: I don't think we need to add the issue here, the commit will have the reference to the PR/issue.

File: attributioncode/validator.go [hunk 2]
```diff
@@ -73,6 +74,7 @@ type Code struct {
 	UA            string
 	ClientID      string
 	SessionID     string
+	DlSource      string
```
Comment (willdurand) [inline attributioncode/validator.go:77]: What do you think about using a slightly longer attribute name?

```suggestion
	DownloadSource      string
```

Comment (bhearsum) [reply attributioncode/validator.go:77]: Up to you, as far as I'm concerned.

**Comment by willdurand** [comment]
> I couldn't get it up and running locally (it seemed like it needed creds to an AWS or GCP database or something? Maybe I misunderstood.)

Did you see these instructions before? https://github.com/mozilla-services/stubattribution/tree/master/testing/stubtestclient#how-to-run-the-stub-service-locally

**Review by willdurand — APPROVED** [review]
Looks good, thanks!

**Pull request merged**